File: app/src/test/java/com/mmutert/freshfreezer/notification/NotificationHelperTest.java
package com.mmutert.freshfreezer.notification;

import com.mmutert.freshfreezer.notification.NotificationHelper;

import org.joda.time.DateTime;
import org.junit.Test;

import org.joda.time.LocalDateTime;
import java.util.concurrent.TimeUnit;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;


public class NotificationHelperTest {

    @Test
    public void testCalculateOffset() {

        LocalDateTime startDate = new LocalDateTime(2020, 7, 15, 14, 0);
        LocalDateTime goalDate = new LocalDateTime(2020, 7, 16, 14, 0);

        long offset = NotificationHelper.calculateOffset(TimeUnit.HOURS, startDate, goalDate);
        assertTrue(offset >= 23);
        assertTrue(offset <= 25);
        assertEquals("Calculated offset does not match actual offset", 24, offset);

        offset = NotificationHelper.calculateOffset(TimeUnit.SECONDS, startDate, goalDate);
        assertEquals("Calculated offset does not match actual offset", 86400, offset);
    }
}

File: app/src/test/java/com/mmutert/freshfreezer/notification/NotificationHelperTest.kt
package com.mmutert.freshfreezer.notification

import com.mmutert.freshfreezer.notification.NotificationHelper.calculateOffset
import org.joda.time.LocalDateTime
import org.junit.Assert
import org.junit.Test
import java.util.concurrent.TimeUnit

class NotificationHelperTest {
    @Test
    fun testCalculateOffset() {
        val startDate = LocalDateTime(2020, 7, 15, 14, 0)
        val goalDate = LocalDateTime(2020, 7, 16, 14, 0)
        var offset = calculateOffset(TimeUnit.HOURS, startDate, goalDate)
        Assert.assertTrue(offset >= 23)
        Assert.assertTrue(offset <= 25)
        Assert.assertEquals("Calculated offset does not match actual offset", 24, offset)
        offset = calculateOffset(TimeUnit.SECONDS, startDate, goalDate)
        Assert.assertEquals("Calculated offset does not match actual offset", 86400, offset)
    }
}